### DISCUSSION TITLE
[Auth] ReCaptcha Flow and restricted keys

### DISCUSSION THREAD

**Discussion by KDederichs** [firebase/firebase-ios-sdk#8304] [Q&A] [answered]
Hey,
I'm running into an issue (mainly on emulators) where Phone Auth will redirect to the ReCaptcha Web flow.
The problem here is that it uses the iOS and Android specific keys which I'd like to restrict to the application itself for obvious reasons.
Since those keys are bound to the app the web recaptcha flow will fail though, since it's not executing them through the app but to through a browser and as a result I'll get a 403 - Requests from this application blocked error.

Is there a solution to this except removing the application restrictions from the keys?

**Comment by morganchen12** [comment, marked as answer]
iOS Simulator can't receive the APNs message that Phone Auth on-device uses to verify the device, which is why it needs to use recaptcha. If you want to avoid loosening your API key restrictions, you can opt to do all of your testing on-device, but otherwise afaik you'll have to change your key restrictions. cc @rosalyntan please double-check my answer

**Reply by KDederichs** [reply]
Ok thanks, might be worth mentioning in the documentation somewhere (if it is I didn't find it) :)
